pocs/tests/test_rs232.py

```diff
@@ -43,34 +68,51 @@ def test_basic_no_op(handler):
     # Open is automatically called by SerialData.
     assert ser.is_connected
 
-    # no_op handler doesn't do any reading, analogous to /dev/null, which
-    # never produces any output.
-    assert '' == ser.read(retry_delay=0.01, retry_limit=2)
-    assert 0 == ser.write('abcdef')
-
-    # Disconnect from the serial port.
+    # connect() is idempotent.
+    ser.connect()
     assert ser.is_connected
-    ser.disconnect()
-    assert not ser.is_connected
 
-    # Should no longer be able to read or write.
-    with pytest.raises(AssertionError):
-        ser.read(retry_delay=0.01, retry_limit=1)
-    with pytest.raises(AssertionError):
-        ser.write('a')
+    # Several passes of reading, writing, disconnecting and connecting.
+    for _ in range(3):
+        # no_op handler doesn't do any reading, analogous to /dev/null, which
+        # never produces any output.
+        assert '' == ser.read(retry_delay=0.01, retry_limit=2)
+        assert b'' == ser.read_bytes(size=1)
+        assert 0 == ser.write('abcdef')
```

Comment:
So does `write` ever return a valid number or is it always 0?

Reply:
The no_op handler never returns anything, but other handles do report the size written.